### ISSUE TITLE
Pulling from base overwrites the root namespace

### ISSUE THREAD

**Issue by runarorama** [unisonweb/unison#1488]
```
.> ls

  1. _crs/    (207 definitions)
  2. base/    (749 definitions)
  3. lib/     (22 definitions)
  4. mybase/  (1535 definitions)
  5. runar/   (10 definitions)
  6. scratch/ (1 definition)
  7. work/    (48 definitions)

.> pull https://github.com/unisonweb/base:.trunk base

  Nothing changed as a result of the merge.


  😶

  base was already up-to-date with https://github.com/unisonweb/base:.trunk.

.>
.> ls

  1. _crs/    (207 definitions)
  2. base/    (749 definitions)
  3. lib/     (22 definitions)
  4. mybase/  (1535 definitions)
  5. runar/   (10 definitions)
  6. scratch/ (1 definition)
  7. work/    (48 definitions)

.> exit
~/w/unison (master|✚1) ❯❯❯ stack exec unison

   _____     _
  |  |  |___|_|___ ___ ___
  |  |  |   | |_ -| . |   |
  |_____|_|_|_|___|___|_|_|

  Welcome to Unison!

  I'm currently watching for changes to .u files under ~/work/unison

  Type help to get help. 😎

.> ls

  1.  ()/              (0 definitions)
  2.  .                ((b ->{𝕖} c) -> (a ->{𝕖} b) -> a ->{𝕖} c)
  3.  <|               ((a ->{𝕖} b) -> a ->{𝕖} b)
  4.  Abort            (type)
  5.  Abort/           (6 definitions)
  6.  Ask              (type)
  7.  Ask/             (4 definitions)
  8.  Author           (type)
  9.  Author/          (8 definitions)
  10. Boolean          (builtin type)
```

It says "nothing changed as a result", but actually my whole codebase is gone after restarting ucm.

**Comment by aryairani** [comment]
damn, that happened to me too.  ref #1484

**Comment by aryairani** [comment]
@runarorama Thanks for this clue; you should be able to restore your codebase using the reflog

**Comment by runarorama** [comment]
I can reproduce reliably, fyi

**Comment by aryairani** [comment]
@runarorama Could you link a transcript as a gist?  That was gonna be my next step, but I haven't done it yet.  It's just for fix development purposes, so it can be slow; do pulls or whatever.

**Comment by runarorama** [comment]
Transcript is just:

```ucm
.> pull https://github.com/unisonweb/base:.trunk base
```

**Comment by runarorama** [comment]
The issue doesn't manifest until you restart UCM, so I'm not sure we can test this easily with a transcript.

**Comment by aryairani** [comment]
The git tests have an example of testing things that are hard to test with a transcript

**Comment by aryairani** [comment]
> Transcript is just:
> 
> ```
> .> pull https://github.com/unisonweb/base:.trunk base
> ```

Sorry, I just meant anything that visibly demonstrates reproducing the issue, starting with `ucm init`, because I can't reproduce it reliably:

```
lrrr:tmp arya$ ucm -version
ucm version: release/M1l
lrrr:tmp arya$ ucm -codebase c init
Initializing a new codebase in: /private/tmp/c
lrrr:tmp arya$ ucm -codebase c

   _____     _             
  |  |  |___|_|___ ___ ___ 
  |  |  |   | |_ -| . |   |
  |_____|_|_|_|___|___|_|_|
  
  Welcome to Unison!
  
  I'm currently watching for changes to .u files under /private/tmp
  
  Type help to get help. 😎
  
  Enter pull https://github.com/unisonweb/base:.releases._M1l .base to set up the default base
  library. 🏗

.> pull https://github.com/unisonweb/base:.trunk base
                                                               
  Here's what's changed in base after the merge:
  
  Added definitions:
  
    1.   ability Abort (+2 metadata)
    2.   ability Ask a (+2 metadata)
    3.   unique type Author
    4.   builtin type Boolean
[less] Use space/arrow keys to navigate, or 'q' to return to ucm
q
.> ls

  1. base/ (753 definitions)

.> 
lrrr:tmp arya$ ucm -codebase c

   _____     _             
  |  |  |___|_|___ ___ ___ 
  |  |  |   | |_ -| . |   |
  |_____|_|_|_|___|___|_|_|
  
  Welcome to Unison!
  
  I'm currently watching for changes to .u files under /private/tmp
  
  Type help to get help. 😎

.> ls

  1. base/ (753 definitions)

.> 
```

**Comment by aryairani** [comment]
Closed by #1501
